packages/react-form-state/src/FormState.ts

```diff
@@ -186,52 +180,106 @@ export default class FormState<
   }
 
   @memoize()
-  private getHandlersForField(path: keyof FieldStates<Fields>) {
+  @bind()
+  private fieldWithHandlers<Key extends keyof Fields>(
+    field: FieldStates<Fields>[Key],
+    fieldPath: Key,
+  ) {
     return {
-      onChange: this.updateField.bind(this, path),
-      onBlur: this.blurField.bind(this, path),
+      ...(field as FieldState<Fields[Key]>),
```

Comment:
do we not need the `name: path` piece any more?

Reply:
Added back